bolt/include/bolt/Passes/PAuthGadgetScanner.h

```diff
@@ -199,19 +199,34 @@ struct Report {
   virtual void generateReport(raw_ostream &OS,
                               const BinaryContext &BC) const = 0;
 
+  virtual const ArrayRef<MCPhysReg> getAffectedRegisters() const { return {}; }
+  virtual void
+  setOverwritingInstrs(const std::vector<MCInstReference> &Instrs) {}
+
   void printBasicInfo(raw_ostream &OS, const BinaryContext &BC,
                       StringRef IssueKind) const;
 };
 
 struct GadgetReport : public Report {
   const GadgetKind &Kind;
+  SmallVector<MCPhysReg> AffectedRegisters;
   std::vector<MCInstReference> OverwritingInstrs;
```

Comment:
I'm guessing that for consistency's sake, `AffectedRegisters` and `OverwritingInstrs` probably both should be `SmallVector` or `std::vector`?

Reply:
I agree that having two different implementations of vector is quite strange. On the other hand, these two variables are not as consistent as they could look like - there is seemingly similar relationship between `State::LastInstWritingReg` and `PacRetAnalysis::RegsToTrackInstsFor` (where N registers correspond to exactly N elements of `LastInstWritingReg` in the same order). Though, using the same vector implementation and adding a comment is definitely better :)

Maybe replacing both vectors with sets would be even better.

Reply:
Kept these fields vectors (but both are `SmallVector`s now), added comments. With `SmallVector` one could return `SmallVectorImpl &` without hard-coding the size parameter or even return `ArrayRef` (which handles `std::vector` as well), but there doesn't seem to be similar option for `SmallSet`.

Reply:
Thanks for looking in more detail and adapting the code. Looks good!